dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```
@@ -825,6 +846,31 @@ private GHRateLimit getGhRateLimitQuietly() {
         return startRateLimit;
     }
 
+    private GHRef[] getBranchesAndTags(GHRepository repo) throws IOException {
+        boolean getBranchesSucceeded = false;
+        GHRef[] branches = {};
+        GHRef[] tags = {};
+
+        // getRefs() fails with a GHFileNotFoundException if there are no matching results instead of returning an empty array/null.
+        try {
+            branches = repo.getRefs("refs/heads/");
+            getBranchesSucceeded = true;
+        } catch (GHFileNotFoundException ex) {
+            LOG.debug("No branches found for " + repo.getName(), ex);
+        }
+
+        try {
+            tags = repo.getRefs("refs/tags/");
+        } catch (GHFileNotFoundException ex) {
+            LOG.debug("No tags found for  " + repo.getName());
+            if (!getBranchesSucceeded) {
+                LOG.info("Repo: " + repo.getName() + "has no branches or tags.");
```

I think you should throw an exception here.